lib/data_explorer_query_group_bookmarkable.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+class DataExplorerQueryGroupBookmarkable < BaseBookmarkable
+  def self.model
+    DataExplorer::QueryGroup
+  end
+
+  def self.serializer
+    UserDataExplorerQueryGroupBookmarkSerializer
+  end
+
+  def self.preload_associations
+    [:data_explorer_queries, :groups]
+  end
+
+  def self.list_query(user, guardian)
+    group_ids = user.visible_groups.pluck(:id) 
```

Comment:
An optimization you can do here is to not even fetch the visible_groups if the user is admin:

```
group_ids = []
if !user.admin?
  group_ids = user.visible_groups.pluck(:id)
  return if group_ids.empty?
end
```

Reply:
You're right. Done.